astropy/table/column.py

```diff
@@ -1230,7 +1237,9 @@ def __setitem__(self, index, value):
 
         # update indices
         self.info.adjust_indices(index, value, len(self))
-        ma.MaskedArray.__setitem__(self, index, value)
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore', MaskedArrayFutureWarning)
```

Comment:
I just did before I saw your reply. `if ... else ...` is there, except I just use the warning variable itself as a switch; Is that acceptable to you?